spec/amazon_ses_mailer/base_spec.rb

```diff
@@ -1,79 +1,57 @@
 # frozen_string_literal: true
 
+require 'my_interceptor'
+
 RSpec.describe AmazonSesMailer::Base do
-    let(:template_name) { 'welcome_email' }
-    let(:options) { { template: template_name } }
-    let(:delivery_method) { nil }
-    subject { described_class.new(template_name) }
-  
-    describe '.mail' do
-      before do
-        allow(AmazonSesMailer::Base).to receive(:delivery_method)
-          .and_return(delivery_method)
-      end
-  
-      it 'creates a new message' do
-        expect(AmazonSesMailer::Message).to receive(:new)
-        subject.mail(options)
+  let(:template_name) { 'welcome_email' }
+  let(:options) { { template: template_name } }
+  let(:delivery_method) { nil }
+  subject { described_class.new(template_name) }
+
+  describe '.deliveries' do
+    context 'when there is no email delivered' do
+      it 'should have zero email deliveries' do
+        expect(AmazonSesMailer::Base.deliveries).to eq([])
       end
     end
-  
-    describe '.deliveries' do
-      context 'when there are no deliveries' do
-        it 'should return empty array' do
-          expect(AmazonSesMailer::Base.deliveries).to eq([])
-        end
-      end
-  
-      context 'when there are deliveries' do
-        let(:delivery_method) { :test }
-        before do
-          allow(AmazonSesMailer::Base).to receive(:delivery_method)
-            .and_return(delivery_method)
-        end
-        it {
-          expect { AmazonSesMailer::Base.mail(options).deliver }
-            .to change { AmazonSesMailer::Base.deliveries.count }.by(1)
-        }
+
+    context 'when an email is delivered' do
+      options = { from: 'user@example.com', to: 'user@example.com' }
+      AmazonSesMailer::Base.delivery_method = :test
+      it 'should not have zero email deliveries' do
+        AmazonSesMailer::Base.mail(options).deliver
```

Comment:
This is not a unit test and the `not_to` is not an accurate way to assert values.